components/DropdownMenu.jsx

```diff
@@ -0,0 +1,86 @@
+import { useState, useRef, useEffect } from "react";
+import styled from "styled-components";
+
+export default function DropdownMenu({
+  options,
+  selectedOption,
+  onOptionSelect,
+}) {
+  const [isOpen, setIsOpen] = useState(false);
+  const dropdownRef = useRef(null);
+
+  function handleToggle() {
+    setIsOpen(!isOpen);
+  }
+
+  function handleOptionSelect(option) {
+    onOptionSelect(option);
+    setIsOpen(false);
+  }
+
+  function handleClickOutside(event) {
+    if (dropdownRef.current && !dropdownRef.current.contains(event.target)) {
+      setIsOpen(false);
+    }
+  }
+
+  useEffect(() => {
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, []);
+
+  return (
+    <DropdownContainer ref={dropdownRef}>
+      <DropdownButton onClick={handleToggle}>{selectedOption}</DropdownButton>
+      {isOpen && (
+        <DropdownMenuContainer>
+          {options.map((option, index) => (
+            <DropdownMenuItem
+              key={index}
```

Comment:
I thought we were not allowed to use**index** as a key.

Reply:
This is true. but as we are just using a fixed amount of options so the index as key is fine